python/ml4ir/base/model/architectures/dnn.py

```diff
@@ -11,12 +17,30 @@ class DNNLayer:
 
 
 class DNN:
-    def __init__(self, model_config):
-        self.layer_ops: List = self.define_architecture(model_config)
+    def __init__(self, model_config: dict, feature_config: dict, file_io):
+        self.file_io: FileIO = file_io
+        self.layer_ops: List = self.define_architecture(model_config, feature_config)
+
+    def define_architecture(self, model_config: dict, feature_config: dict):
+        """
+        Convert the model from model_config to a List of tensorflow.keras.layer
 
-    def define_architecture(self, model_config):
+        :param model_config: dict corresponding to the model config
+        :param feature_config: dict corresponding to the feature config, only used in case of classification if the last
+            layer of the model_config doesn't have a units number defined (or set to -1). In which case we retrieve the
+            label vocabulary defined in the feature_config to deduce the number of units.
+        :return: List[layers]: list of keras layer corresponding to each of the layers defined in the model_config.
+        """
         def get_op(layer_type, layer_args):
             if layer_type == DNNLayer.DENSE:
+                if not "units" in layer_args or layer_args["units"] == -1:
+                    try:
+                        label_feature_info = feature_config.get_label()
+                        vocabulary_keys, vocabulary_ids = get_vocabulary_info(label_feature_info, self.file_io)
+                        layer_args["units"] = len(vocabulary_keys) + OOV
+                    except:
+                        raise KeyError("units can only not be specified for the last layer, in which can a vocabulary"
```

Comment:
I would suggest a Keras alike message, a bit more actionable
"We were not able to find information for the output layer of your DNN. Try specifying the number of output units either by passing "units" in the model configuration yaml file or units in the feature configuration file."